Add example readme (#1)
DevExpress-Examples/reporting-blazor-email-report: README.md

<!-- default badges list -->
[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/T1232536)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->
# Reporting for Blazor - Email a Report from the Native Blazor Report Viewer

This example uses the [Mailkit](https://mimekit.net/docs/html/Introduction.htm) library to send an email from our Native Blazor Report Viewer.

> The **Send Email** button in the Viewer’s toolbar opens the **Send Email** dialog ([DxPopup](https://docs.devexpress.com/Blazor/DevExpress.Blazor.DxPopup)). In the dialog,  you can specify the recipient list, subject, attachment, and body. Click the **Send** button to send the report with the specified settings:

Reviewer: ```suggestion
The **Send Email** button in the Viewer’s toolbar opens the **Send Email** dialog ([DxPopup](https://docs.devexpress.com/Blazor/DevExpress.Blazor.DxPopup)). In the dialog,  you can specify the recipient list, subject, attachments, and body. Click the **Send** button to send the report with the specified settings:
```

Author: @dirk-pieterse In this example, there can be only one file in the attachment. I am not sure about plural here.

![Report Viewer - Send Email Window](images/send-email-window.png)

> The resulting email may look as follows:

Reviewer: ```suggestion
An example of the resulting email:
```

![Report Viewer - Sent Email Example](images/sent-email-example.png)

## Implementation Details

### UI Elements

> The example handles the [`CustomizeToolbar`](https://docs.devexpress.com/XtraReports/DevExpress.Blazor.Reporting.DxReportViewer.OnCustomizeToolbar) event to add a **Send Email** button to the Viewer's Toolbar. The code snippet below adds locates the **Export To** command and adds the new button next to it:

Reviewer: `adds locates` It adds or it locates?

Author: ```suggestion
The example handles the [`CustomizeToolbar`](https://docs.devexpress.com/XtraReports/DevExpress.Blazor.Reporting.DxReportViewer.OnCustomizeToolbar) event to add a **Send Email** button to the Viewer's Toolbar. The code snippet below locates the **Export To** command and adds the new button next to it:
```

```cs
void OnCustomizeToolbar(ToolbarModel toolbarModel) {
    toolbarModel.AllItems.Insert(toolbarModel.AllItems.FindIndex(i => i.Id == ToolbarItemId.ExportTo), new ToolbarItem() {
        IconCssClass = "mail-icon",
        Text = "Send Email",
        AdaptiveText = "Send Email",
        AdaptivePriority = 1,
        Click = (args) => {
            IsPopupVisible = ValidateEditingFields();
            return Task.CompletedTask;
        }
    });
}
```

> A click on the newly added button opens a [DxPopup](https://docs.devexpress.com/Blazor/DevExpress.Blazor.DxPopup) where users can specify email options: recipients, subject, body. The popup form uses the following components:

Reviewer: ```suggestion
A click on the newly added button opens a [DxPopup](https://docs.devexpress.com/Blazor/DevExpress.Blazor.DxPopup) where users can specify email options: recipients, subject, and body. The pop-up form uses the following components:
```

- [DxTagBox](https://docs.devexpress.com/Blazor/DevExpress.Blazor.DxTagBox-2) allows users to select recipients the list of the recipients.
- [DxTextBox](https://docs.devexpress.com/Blazor/DevExpress.Blazor.DxTextBox) allows users specify the email subject and attachment file name.
- [DxComboBox](https://docs.devexpress.com/Blazor/DevExpress.Blazor.DxComboBox-2) allows users to select attachment format.
- [DxHtmlEditor](https://docs.devexpress.com/Blazor/DevExpress.Blazor.DxHtmlEditor?v=24.1) allows users to specify the mail body.

For the `DxPopup` configuration in this example, refer to the following file: [ReportViewer.razor](BlazorReportViewer/Pages/ReportViewer.razor#L28).

The [DxToastProvider](https://docs.devexpress.devx/Blazor/DevExpress.Blazor.DxToastProvider?v=24.1) component displays toast notifications to users. 
For the `DxToastProvider` configuration, refer to the following file: [ReportViewer.razor](BlazorReportViewer/Pages/ReportViewer.razor#L73).

### Email Service 

> [!WARNING]  

> > In this example, we specify credentials for the authenticate to a SMTP server. In production, we recommend that you use [secrets](https://learn.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-8.0&tabs=windows) to store sensitive information.

Reviewer 2: authentication

Author: ```suggestion
> In this example, we specify credentials for the authentication to a SMTP server. In production, we recommend that you use [secrets](https://learn.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-8.0&tabs=windows) to store sensitive information.
```
